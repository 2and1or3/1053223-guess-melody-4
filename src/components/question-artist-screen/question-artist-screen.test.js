import React from "react";
import renderer from "react-test-renderer";
import QuestionArtistScreen from './question-artist-screen.jsx';

const question = {
  type: `artist`,
  artist: `true-person`,
  trackSrc: `src-for-track`,
  answers: [
    {
      artist: `true-person`,
      pictureSrc: `http://placehold.it/134x134`,
    },
    {
      artist: `person-2`,
      pictureSrc: `http://placehold.it/134x134`,
    },
    {
      artist: `person-3`,
      pictureSrc: `http://placehold.it/134x134`,
    },
  ]
};

it(`Render QuestionArtistScreen component`, () => {
  const tree = renderer
    .create(<QuestionArtistScreen
      question = {question}
      onAnswer = {() => {}}
      renderPlayer = {() => {}}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
