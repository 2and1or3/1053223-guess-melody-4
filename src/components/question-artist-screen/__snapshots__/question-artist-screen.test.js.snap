// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render QuestionArtistScreen component 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx="390"
        cy="390"
        r="370"
        style={
          Object {
            "filter": "url(#blur)",
            "transform": "rotate(-90deg) scaleY(-1)",
            "transformOrigin": "center",
          }
        }
      />
    </svg>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      className="game__track"
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          type="button"
        />
        <div
          className="track__status"
        >
          <audio
            src="src-for-track"
          />
        </div>
      </div>
    </div>
    <form
      className="game__artist"
    >
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="true-person0"
          name="answer"
          onChange={[Function]}
          type="radio"
          value="true-person"
        />
        <label
          className="artist__name"
          htmlFor="true-person0"
        >
          <img
            alt="true-person"
            className="artist__picture"
            src="http://placehold.it/134x134"
          />
          true-person
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="person-21"
          name="answer"
          onChange={[Function]}
          type="radio"
          value="person-2"
        />
        <label
          className="artist__name"
          htmlFor="person-21"
        >
          <img
            alt="person-2"
            className="artist__picture"
            src="http://placehold.it/134x134"
          />
          person-2
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="person-32"
          name="answer"
          onChange={[Function]}
          type="radio"
          value="person-3"
        />
        <label
          className="artist__name"
          htmlFor="person-32"
        >
          <img
            alt="person-3"
            className="artist__picture"
            src="http://placehold.it/134x134"
          />
          person-3
        </label>
      </div>
    </form>
  </section>
</section>
`;
